---
title: "gke/ERR/2021_011"
linkTitle: "ERR/2021_011"
weight: 1
type: docs
description: >
  ip-masq-agent not reporting errors
---

**Product**: [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine)\
**Rule class**: ERR - Something that is very likely to be wrong

### Description

If ip-masq-agent is reporting errors, it is possible that the config received
is invalid. In that case, it is possible that the applied config is not
reflecting the desired masquerading behavior, which could lead to unexpected
connectivity issues.

### Remediation

For more details on how to configure ip-masq-agent, please review:

You can use the following Cloud Logging filter to find matching log lines:

```
resource.type="k8s_container"
resource.labels.container_name="ip-masq-agent"
log_id("stderr")
```

### Further information

- [ip-masq-agent configuration](https://cloud.google.com/kubernetes-engine/docs/how-to/ip-masquerade-agent#config_agent_configmap)
