---
title: "gke/ERR/2021_005"
linkTitle: "ERR/2021_005"
weight: 1
type: docs
description: >
  GKE nodes aren't reporting connection issues to storage.google.com.
---

**Product**: [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine)\
**Rule class**: ERR - Something that is very likely to be wrong

### Description


GKE node need to download artifacts from storage.google.com:443 when
booting. If a node reports that it can't connect to storage.google.com,
it probably means that it can't boot correctly.

### Remediation

You can use the following filter to find matching log lines:

```
resource.type="k8s_node"
log_id("kubelet")
jsonPayload.MESSAGE:"Failed to connect to storage.googleapis.com"
```

### Further information

- [GKE
  Troubleshooting](https://cloud.google.com/kubernetes-engine/docs/troubleshooting)
